bugreportutils/bugreportutils.go

```diff
@@ -43,35 +43,63 @@ var (
-
-	// sensorLineMMinusRE is a regular expression to match the sensor list line in the sensorservice dump of a bug report from MNC or before.
-	sensorLineMMinusRE = regexp.MustCompile(`(?P<sensorName>[^|]+)` + `\|` + `(?P<sensorManufacturer>[^|]+)` + `\|` +
-		`(\s*version=(?P<versionNumber>\d+)\s*\|)?` + `\s*(?P<sensorTypeString>[^|]+)` +
-		`\|` + `\s*(?P<sensorNumber>0x[0-9A-Fa-f]+)`)
-
-	// sensorLineNPlusRE is a regular expression to match the sensor list line in the sensorservice dump of a bug report starting from NRD42 and onwards.
-	sensorLineNPlusRE = regexp.MustCompile(`(?P<sensorNumber>0x?[0-9A-Fa-f]+)` + `\)\s*` +
+	pidRE = regexp.MustCompile(`PID #` + `(?P<pid>\d+)` + `: ProcessRecord[^:]+:` +
+		`(?P<app>[^/]+)` + `/` + `(?P<uid>.*)` + `}`)
+
+	// sensorLineMMinusRE is a regular expression to match the sensor list line in the
+	// sensorservice dump of a bug report from MNC or before.
+	sensorLineMMinusRE = regexp.MustCompile(`(?P<sensorName>[^|]+)` + `\|` +
+		`(?P<sensorManufacturer>[^|]+)` + `\|` + `(\s*version=(?P<versionNumber>\d+)\s*\|)?` +
+		`\s*(?P<sensorTypeString>[^|]+)` + `\|` + `\s*(?P<sensorNumber>0x[0-9A-Fa-f]+)\s*` + `\|` +
+		`.*` + `\|` + `\s*type=\d+\s*\|` + `\s*(?P<requestMode>[^|]+)` + `\s*\|` +
+		`(?P<variableOne>[^|]+)` + `\s*\|` + `(?P<variableTwo>[^|]+)` + `\s*\|` +
+		`(?P<batching>[^|]+)` + `\s*\|` + `(?P<wakeUp>[^|]+)`)
+
+	// sensorLineOneRE is a regular expression to match the first line of sensor information
+	// from the sensor list line in the sensorservice dump in the bugreport starting
+	// from NRD42 and onwards.
+	sensorLineOneRE = regexp.MustCompile(`(?P<sensorNumber>0x?[0-9A-Fa-f]+)` + `\)\s*` +
 		`(?P<sensorName>[^|]+)` + `\s*\|` + `(?P<sensorManufacturer>[^|]+)` + `\|\s*ver:\s*` +
-		`(?P<versionNumber>\d+)` + `\s*\|\s*type:\s*` + `(?P<sensorTypeString>[^(]+)` + `\(\d+\)\s*\|`)
+		`(?P<versionNumber>\d+)` + `\s*\|\s*type:\s*` + `(?P<sensorTypeString>[^(]+)` +
+		`\(\d+\)\s*\|` + `\s*perm:\s*` + `(?P<sensorPerm>[^|]+)`)
+
+	// sensorLineTwoRE is a regular expression to match the second line of sensor information
+	// from the sensor list line in the sensorservice dump in the bugreport starting
+	// from NRD42 and onwards.
+	sensorLineTwoRE = regexp.MustCompile(`\s*(?P<requestMode>[^|]+)` + `\s*\|` +
+		`(?P<variableOne>[^|]+)` + `\s*\|` + `(?P<variableTwo>[^|]+)` + `\s*\|` +
+		`(?P<batching>[^|]+)` + `\s*\|` + `(?P<wakeUp>[^|]+)`)
+
+	// BatchingDataRE is a regular expression that matches the max and reserved data quantity
+	// from the sensor infomration listed in the sensorservice dump of a bug report.
+	// e.g FIFO (max,reserved) = (10000, 3000) events
+	BatchingDataRE = regexp.MustCompile(`FIFO\s*\(max,reserved\)\s*=\s*\(` + `(?P<maxNum>\d+)` +
+		`,` + `(?P<reservedNum>[^|]+)` + `\) events\s*`)
+
+	rateRE    = regexp.MustCompile(`=(?P<rateVal>[\-\+]?[0-9]*(\.[0-9]+)+)Hz`)
```

Comment:
A comment stating that rateRE, delayRE & fifoMAxRE are to parse the sensor information from the dumpsys.